src/components/NavBar.tsx

```diff
@@ -1,37 +1,96 @@
 import { Link } from 'react-router-dom';
-import logo from './../assets/logo.png';
+import { useEffect } from 'react';
 import useUserStore from '../stores/useUserStore';
 
 function NavBar() {
   const { isLoggedIn } = useUserStore();
 
+  useEffect(() => {
+    if (typeof window === 'undefined') return;
+
+    if (
+      localStorage.theme === 'dark' ||
+      (!('theme' in localStorage) &&
+        window.matchMedia('(prefers-color-scheme: dark)').matches)
+    ) {
+      document.documentElement.classList.add('dark');
+    } else {
+      document.documentElement.classList.remove('dark');
+    }
+  }, []);
+
+  function toggleTheme() {
+    const html = document.documentElement;
+    const isDark = html.classList.toggle('dark');
+    localStorage.theme = isDark ? 'dark' : 'light';
+  }
+
   return (
-    <div className="navbar-hld">
-      <Link to="/" className="navbar__logo">
-        <img className="logo" src={logo} height="40px" />
+    <header className="site-top">
+      <Link to="/" className="site-top__logo">
+        <svg
+          className="logo__img"
+          aria-hidden="true"
+          width="28"
+          height="28"
+          viewBox="0 0 24 24"
+        >
+          <path d="M12 2L2 7l10 5l10-5zm0 7L2 4v13l10 5zm0 0l10-5v13l-10 5z" />
+        </svg>
+        <span className="logo__title">CADR</span>
```

Comment:
na to mysle ze mozna zrobic oddzielny komponent. z logo pewnie bedziemy jeszcze gdzies korzystac w aplikacji